doc/sql.extensions/README.blob_util.md

```diff
@@ -0,0 +1,184 @@
+# `RDB$BLOB_UTIL` package (FB 5.0)
+
+Before Firebird 5 BLOBs could be created appending data using the concatenate (`||`) operator with another BLOB.
+
+However there is two big problems with this approach:
+- BLOBs are always created in database page space, even if that is not required (not going to a persistent table).
+- Every time a concatenation happens, a new BLOB is created, filling the database with garbage.
+
+`RDB$BLOB_UTIL` package now exists to fix that problems.
+
+It allows BLOB creation in temporary page space (as well in the persistent page space) and has a workflow so that a BLOB is created and could have its data incrementally appended before it's closed and be available as a BLOB value. This is much like the BLOB client API works, but now available in PSQL.
+
+## Function `NEW`
+
+`RDB$BLOB_UTIL.NEW` is used to create a new BLOB. It returns a handle (an integer bound to the transaction) that should be used with the others functions of the package.
+
```

Comment:
A blob id is used in the client with a handle. A handle in this context is an id more the blb class inside the engine. A blb has information like current position. RDB$BLOB_UTIL handles model this concept in PSQL.

A blob id for this would be very confusing. Many different variables would have the same id so how one could have multiple parallel seek/read in the same blob id?

Also a blob id is implicitely copied depending on blob charset when they are passed as arguments.